Gems/ROS2/Code/Include/ROS2/Lidar/LidarRaycasterBus.h

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/Component/EntityId.h>
+#include <AzCore/EBus/EBus.h>
+#include <AzCore/Math/Transform.h>
+#include <AzCore/Math/Vector3.h>
+
+namespace ROS2
+{
+    //! Class used for creating typesafe Uuid types.
+    //! It utilizes the phantom types technique.
+    template<typename Tag>
+    class StronglyTypedUuid
+    {
+    public:
+        StronglyTypedUuid() = default;
+        StronglyTypedUuid(AZ::Uuid value)
+            : m_uuid(value)
+        {
+        }
```

Comment:
This constructor should be marked explicit.
In order for this class to be strongly-typed it shouldn't allow implicit conversion from a `Uuid` to a `StronglyTypedUuid<Tag>` since it will allow for untended conversions to StronglyTypedUuids from plain Uuids.
```suggestion
        explicit StronglyTypedUuid(AZ::Uuid value)
            : m_uuid(value)
        {
        }
```